.gitlab-ci.yml

```diff
@@ -0,0 +1,81 @@
+stages:
+  - info
+  - lint
+  - test
+
+ENV info:
+  stage: info
+  tags:
+    - mac-ventura-preview
+  script:
+    - system_profiler SPSoftwareDataType # system info
+    - xcodebuild -version
+    - xcode-select -p # default Xcode
+    - ls /Applications/ | grep Xcode # other Xcodes
+    - xcodebuild -workspace "Datadog.xcworkspace" -scheme "DatadogCore iOS" -showdestinations -quiet # installed iOS destinations
+    - xcodebuild -workspace "Datadog.xcworkspace" -scheme "DatadogCore tvOS" -showdestinations -quiet # installed tvOS destinations
+    - xcbeautify --version
+    - swiftlint --version
+    - carthage version
+    - gh --version
+    - brew -v
+    - bundler --version
+    - python3 -V
+
+Lint:
+  stage: lint
+  tags:
+    - mac-ventura-preview
+  script:
+    - ./tools/lint/run-linter.sh
+    - ./tools/license/check-license.sh
+
+SDK unit tests (iOS):
+  stage: test
+  tags:
+    - mac-ventura-preview
+  variables:
+    TEST_WORKSPACE: "Datadog.xcworkspace"
+    TEST_DESTINATION: "platform=iOS Simulator,name=iPhone 15 Pro Max,OS=17.0.1"
+  script:
+    - make dependencies-gitlab
+    - xcodebuild -workspace "$TEST_WORKSPACE" -destination "$TEST_DESTINATION" -scheme "DatadogCore iOS" -only-testing:"DatadogCoreTests iOS" test | xcbeautify
+    - xcodebuild -workspace "$TEST_WORKSPACE" -destination "$TEST_DESTINATION" -scheme "DatadogCore iOS" -only-testing:"DatadogInternalTests iOS" test | xcbeautify
+    - xcodebuild -workspace "$TEST_WORKSPACE" -destination "$TEST_DESTINATION" -scheme "DatadogCore iOS" -only-testing:"DatadogLogsTests iOS" test | xcbeautify
+    - xcodebuild -workspace "$TEST_WORKSPACE" -destination "$TEST_DESTINATION" -scheme "DatadogCore iOS" -only-testing:"DatadogTraceTests iOS" test | xcbeautify
+    - xcodebuild -workspace "$TEST_WORKSPACE" -destination "$TEST_DESTINATION" -scheme "DatadogCore iOS" -only-testing:"DatadogRUMTests iOS" test | xcbeautify
+    - xcodebuild -workspace "$TEST_WORKSPACE" -destination "$TEST_DESTINATION" -scheme "DatadogCore iOS" -only-testing:"DatadogWebViewTrackingTests iOS" test | xcbeautify
+    - xcodebuild -workspace "$TEST_WORKSPACE" -destination "$TEST_DESTINATION" -scheme "DatadogSessionReplay iOS" test | xcbeautify
+    - xcodebuild -workspace "$TEST_WORKSPACE" -destination "$TEST_DESTINATION" -scheme "DatadogCrashReporting iOS" test | xcbeautify
```

Comment:
food for thought, can you run test for all targets in parallel on different containers.

Something like https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs

Reply:
@ganeshnj that would mean running each as separate GitLab stage, right? We could see what is the overhead of this (for sure the clean build time) vs benefit of parallelization. Total run of unit tests stage is ~2-4minutes, so it might be counter-optimisation.

Reply:
Yes, they will be different runners. 2-4 minutes is good and looks like we don't need any parallel execution.

Ideally there shouldn't be a clean build, the past state must be forwarded to next state.

Reply:
That would be possible with passing artifacts between jobs 👌. Let's keep this in mind. For now doesn't sound necessary so we can start simple.